test/Universe.js

```diff
@@ -6,271 +6,542 @@ import registerSingletons from "./helpers/registerSingletons";
-      "platformTerms",
-    ];
-    for (let ii = 0; ii < kis.length; ii += 1) {
-      const loopi = await universe[getters[ii]]();
-      expect(loopi).to.eq(knownInstances[ii]);
-    }
-  });
+      const kis = [
+        {
+          ki: knownInterfaces.etherToken,
+          addr: "0x498a042f52f1737a77b91dd8107e68d75bf9f478",
+        },
+        {
+          ki: knownInterfaces.euroToken,
+          addr: "0xd102445e80c56d36c7dcd968dc2792b87b236b46",
+        },
+        {
+          ki: knownInterfaces.feeDisbursal,
+          addr: "0xc22b7a02afa706b5bc2fdfc305288bd910812977",
+        },
+        {
+          ki: knownInterfaces.euroLock,
+          addr: "0x147df49452f805d1a35e7ca314f564d1087b112f",
+        },
+        {
+          ki: knownInterfaces.etherLock,
+          addr: "0x1608e04a05080150bbe5d33c95786016287f2990",
+        },
+        {
+          ki: knownInterfaces.gasExchange,
+          addr: "0x42194bf970666c3232f37dc39684a6937a65e2b9",
+        },
+        {
+          ki: knownInterfaces.icbmEtherLock,
+          addr: "0x1d5f6ed3ea6f3417b352d65333ea6eaaf6b1bb1d",
+        },
+        {
+          ki: knownInterfaces.icbmEuroLock,
+          addr: "0x4a882db40af31d2f7ae3cecd67d7a43e8ebf42f5",
+        },
+        {
+          ki: knownInterfaces.identityRegistry,
+          addr: "0x87c308a3f6752feadea2296a0c711ed31aa69c5f",
+        },
+        {
+          ki: knownInterfaces.tokenExchange,
+          addr: "0x147b317c8d0bdf72e3df3395d839af4bc581eebb",
+        },
+        {
+          ki: knownInterfaces.tokenExchangeRateOracle,
+          addr: "0x20216d0d0912d0a44cec2ffbc85043da6a943fda",
+        },
+        {
+          ki: knownInterfaces.platformTerms,
+          addr: "0x20216d0d0912d0a44cec2ffbc85043da6a943fdb",
+        },
+      ];
+      const multiRegTx = await registerSingletons(universe, universeManager, kis);
+      // find if number of events matches
+      const events = multiRegTx.logs.filter(e => e.event === "LogSetSingleton");
+      expect(events.length).to.eq(kis.length);
+      // get all known interfaces
+      const knownInterfacesList = kis.map(({ ki }) => ki);
+      const knownInstances = kis.map(({ addr }) => addr);
+      expect(await universe.getManySingletons(knownInterfacesList)).to.deep.eq(knownInstances);
+      // compare with convenience getters
+      const getters = [
+        "etherToken",
+        "euroToken",
+        "feeDisbursal",
+        "euroLock",
+        "etherLock",
+        "gasExchange",
+        "icbmEtherLock",
+        "icbmEuroLock",
+        "identityRegistry",
+        "tokenExchange",
+        "tokenExchangeRateOracle",
+        "platformTerms",
+      ];
+      for (let ii = 0; ii < kis.length; ii += 1) {
+        const loopi = await universe[getters[ii]]();
+        expect(loopi).to.eq(knownInstances[ii]);
+      }
+    });
 
-  it("should set singleton without convenience method");
+    it("should set singleton without convenience method");
```

Comment:
the point of the test was to set a singleton that does not have convenience method (like universe.neumark()). a singleton where interface is just a random 4 bytes (not a knownInterface)

Reply:
done